backend/btrixcloud/subs.py

```diff
@@ -0,0 +1,345 @@
+"""
+Subscription API handling
+"""
+
+from typing import Callable, Union, Any, Optional, Tuple, List
+import os
+from uuid import UUID
+
+from datetime import datetime
+
+from fastapi import Depends, HTTPException, Request
+import aiohttp
+
+from .orgs import OrgOps
+from .users import UserManager
+from .utils import is_bool
+from .models import (
+    SubscriptionCreate,
+    SubscriptionUpdate,
+    SubscriptionCancel,
+    SubscriptionCreateOut,
+    SubscriptionUpdateOut,
+    SubscriptionCancelOut,
+    Subscription,
+    SubscriptionOut,
+    SubscriptionPortalUrlRequest,
+    SubscriptionPortalUrlResponse,
+    Organization,
+    InviteToOrgRequest,
+    User,
+    UserRole,
+)
+from .pagination import DEFAULT_PAGE_SIZE, paginated_format
+
+
+# if set, will enable this api
+subscriptions_enabled = is_bool(os.environ.get("BILLING_ENABLED"))
+
+
+# if set, will lookup external portalUrl from this endpoint
+external_subs_app_api_url = os.environ.get("BTRIX_SUBS_APP_URL")
+
+
+# ============================================================================
+class SubOps:
+    """API for managing subscriptions. Only enabled if billing is enabled"""
+
+    org_ops: OrgOps
+    user_manager: UserManager
+
+    def __init__(self, mdb, org_ops: OrgOps, user_manager: UserManager):
+        self.subs = mdb["subscriptions"]
+        self.org_ops = org_ops
+        self.user_manager = user_manager
+
+    async def create_new_subscription(
+        self, create: SubscriptionCreate, user: User, request: Request
+    ) -> dict[str, Any]:
+        """create org for new subscription"""
+        subscription = Subscription(
+            subId=create.subId, status=create.status, planId=create.planId
```

Comment:
What is the `subId` that's included in the request body? If this is the payment processor subscription ID, perhaps it can be named `providerSubId` to differentiate from the `Subscription` resource ID?

Reply:
There's only one id, we use a combination of provider subscription id and its opaque in browsertrix but guaranteed to be unique.